examples/spec/integration/initial_search_attributes_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'workflows/upsert_search_attributes_workflow'
+require 'time'
+
+describe 'Temporal::Client.start_workflow', :integration do
+  it 'can start a workflow with initial search attributes' do
+    workflow_id = 'initial_search_attributes_test_wf-' + SecureRandom.uuid
+    expected_binary_checksum = `git show HEAD -s --format=%H`.strip
+
+    initial_search_attributes = {
+      'CustomBoolField' => false,
+      'CustomIntField' => -1,
+      'CustomDatetimeField' => Time.now.utc.iso8601,
+      # These should get overriden when the workflow upserts them
+      'CustomStringField' => 'meow',
+      'CustomDoubleField' => 6.28,
+    }
+    upserted_search_attributes = {
+      'CustomStringField' => 'moo',
+      'CustomDoubleField' => 3.14,
+    }
+    expected_custom_attributes = initial_search_attributes.merge(upserted_search_attributes)
+
+    run_id = Temporal.start_workflow(
+      UpsertSearchAttributesWorkflow,
+      string_value: upserted_search_attributes['CustomStringField'],
+      float_value: upserted_search_attributes['CustomDoubleField'],
```

Comment:
I wonder if not passing bool and int fields here is intentional? Looks like you already have them in the `initial_search_attributes`, so probably missed here?